### DISCUSSION TITLE
[V4] How to use gmail with OAuth2 in Adonis Mail

### DISCUSSION THREAD

**Discussion by stefanozanini** [adonisjs/core#2193] [Help] [answered]
Hi everyone!
I'm trying to migrate my Adonis app from using gmail SMTP directly to OAuth2 (because google is dropping less secure apps), but I can't figure out how to configure it. I know that Adonis Mail provider use Nodemailer internally, and I know that Nodemailer can use OAuth2 to authenticate to the SMTP server but I can't figure out how to configure the Adonis Mail provider to use an OAuth2 token. Is there anyone that has encountered the same problem and can help me figure out how to solve it?

**Comment by thetutlage** [comment, marked as answer]
Hey @stefanozanini . You can use the `oauth2` by defining the following config.

```ts
{
  driver: 'smtp',
  auth: {
    type: 'OAuth2',
    user: '',
    clientId: '',
    clientSecret: '',
  }
}
```

**Reply by stefanozanini** [reply]
@thetutlage thanks for your answer!
It seems to need an access token and a refresh token as well, i followed this guide and i managed to get it working
https://stackoverflow.com/questions/51933601/what-is-the-definitive-way-to-use-gmail-with-oauth-and-nodemailer
